### ISSUE TITLE
Feature: Move column between two other columns

### ISSUE THREAD

**Issue by JackArlington** [swimlane/ngx-datatable#1186]
**I'm submitting a ...**  (check one with "x")
```
[ ] bug report => search github for a similar issue or PR before submitting
[x] feature request
[ ] support request => Please do not submit support request here, post on Stackoverflow or Gitter
```

**Current behavior**

Columns can currently be exchanged by drag&drop. But it is not possible to drop a column between two other columns. 

**Expected behavior**

When a dragged column A is held over the middle of two other columns C and D a small gap opens between C and D. When A is dropped, it is placed between C and D. 

So that the following reordering is possible: A B C D -> B C A D

**Reproduction of the problem**
Try to move a column between two other columns, when reorder is on.

**What is the motivation / use case for changing the behavior?**

The user experience of reordering columns would be improved and it would be much easier to create a desired order. 

**Please tell us about your environment:**

* **Table version:** 11.1.5

* **Angular version:** 5.1.0

* **Browser:** all 

* **Language:** [all | TypeScript X.X | ES6/7 | ES5]

**Comment by arron21** [comment]
Yes Please!

**Comment by htrex** [comment]
Duplicate of #868

**Comment by JackArlington** [comment]
Closed in favor of #868
